### PR TITLE
Upgrade to Angular 16

### PR DESCRIPTION
**Pull request by dariusdv**
Upgrade to Angular 16

Issue https://github.com/spring-petclinic/spring-petclinic-angular/issues/73

### CODE REVIEW

File: src/app/vets/vet-edit/vet-edit.component.ts [hunk 1]
```diff
@@ -26,35 +26,35 @@ import {VetService} from '../vet.service';
...
     this.specList = [] as Specialty[];
     this.buildForm();
   }
 
   buildForm() {
-    this.idCtrl = new FormControl(null);
-    this.firstNameCtrl = new FormControl('', [Validators.required, Validators.minLength(2)]);
-    this.lastNameCtrl = new FormControl('', [Validators.required, Validators.minLength(2)]);
-    this.specialtiesCtrl = new FormControl(null);
+    this.idCtrl = new UntypedFormControl(null);
```
Comment (arey) [inline src/app/vets/vet-edit/vet-edit.component.ts:54]: Do you know if we can easily migrate to typed forms?

**Review by arey — COMMENTED** [review]
Thanks @dariusdv for the upgrade.
I've posted two comments. If you want, we could create PR for each one.
Let me know when you think the migration is done.

File: src/app/vets/vets.module.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@
 import {NgModule} from '@angular/core';
 import {CommonModule} from '@angular/common';
 import {FormsModule, ReactiveFormsModule} from '@angular/forms';
-import {MatSelectModule} from '@angular/material/select';
+import {MatLegacySelectModule as MatSelectModule} from '@angular/material/legacy-select';
```
[the remark this answers is not in the document]
Comment (dariusdv) [reply src/app/vets/vets.module.ts:26]: I will look into it, I didn't try migrate. I think it should be easy to do it.

**Comment by dariusdv** [comment]
Would you recommend to start two new pull requests or to continue with the implementation on this one? If I start new PRs for these updates, then there might be conflicts between the new PRs and the main branch until this PR is merged.

**Comment by dariusdv** [comment]
@arey 
I migrated to the new forms and mat select. 
Everything seems to be ok now with this PR, I think it can be merged.
Let me know if you find anything else.

**Comment by arey** [comment]
Everything looks ok Thank you Darius for your work

**Pull request merged**